src/main/java/com/github/invictum/reportportal/ReportPortalListener.java

```diff
@@ -129,7 +130,10 @@ public void stepFinished(List<ScreenshotAndHtmlSource> list, ZonedDateTime zoned
 
     @Override
     public void testFailed(TestOutcome testOutcome, Throwable throwable) {
-        // Not used by listener
+        TestType testType = TestType.byTestSource(testOutcome.getTestSource());
+        if (testType == TestType.BDD) {
+            testFinished(testOutcome);
+        }
```

Comment:
1. This might be dangerous without extensive testing of all supported bdd style tests.
2. Might worths to add a unit test for this logic.